components/waypoint-movement.js

```diff
@@ -0,0 +1,239 @@
+/*
+Original authors — credit is appreciated but not required:
+
+     2021 — [Florian Isikci] <[florian.isikci@vhiterabbit.com]>
+
+ This is free and unencumbered software released into the public domain.
+ Anyone is free to copy, modify, publish, use, compile, sell, or distribute
+ this software, either in source code form or as a compiled binary, for any
+ purpose, commercial or non-commercial, and by any means.
+
+ In jurisdictions that recognize copyright laws, the author or authors of
+ this software dedicate any and all copyright interest in the software to
+ the public domain. We make this dedication for the benefit of the public
+ at large and to the detriment of our heirs and successors. We intend this
+ dedication to be an overt act of relinquishment in perpetuity of all
+ present and future rights to this software under copyright law.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+WL.registerComponent('waypoint-movement', {
+    /** This object is the container of the waypoints. It should contain children
+    with an alphebetically ascending naming (e.g. A, B, C, D, E, F) */
+    pathObject: {type: WL.Type.Object},
+    /** Movement speed of the object */
+    speed: {type: WL.Type.Float, default: 1.0},
+    /** Distance in normal space [0, 0.5) after which the objects starts moving on a curve */
+    curveDistance: {type: WL.Type.Float, default: 0.1},
+}, {
+    init: function() {
+      this.children = this.pathObject.children.sort(function(a, b) { return a.name.localeCompare(b.name) });
+      this.positions = new Array(this.children.length);
+      for (var i = 0; i < this.children.length; i++) {
```

Comment:
Try to avoid `var` in favor of `let` please :)

[Here's an article on why for anyone interested](https://blog.usejournal.com/awesome-javascript-no-more-var-working-title-999428999994).